src/hermes/commands/deposit/invenio.py

```diff
@@ -111,18 +110,30 @@ def deposit(click_ctx: click.Context, ctx: CodeMetaContext):
     )
     deposit_url = f"{site_url}/{depositions_api_path}"
 
-    if existing_record_url is not None:
-        # TODO: Get by calling new version on latest existing record
-        deposit_url = None
-        raise NotImplementedError(
-            "At the moment, hermes can not create new versions of existing records"
+    deposition_metadata = invenio_ctx["depositionMetadata"]
+    try:
+        latest_metadata = invenio_ctx["latestRecord"]["metadata"]
+        if deposition_metadata.get("version") == latest_metadata.get("version"):
+            raise ValueError("Version already deposited.")
+
+        record_id = invenio_ctx["latestRecord"]["id"]
+    except KeyError:
+        record_id = None
+
+    if record_id is not None:
+        deposit_url += f'/{record_id}/actions/newversion'
+        response = session.post(deposit_url)
```

Comment:
This response needs checking. If the request didn't succeed it was very likely because we are not the owner of this existing record. (I noticed this when trying to use your `record_id` and my `auth_token`.)

Currently we then fail in line 128 with `KeyError: 'links'` on the `old_deposit`.

Reply:
Well, I hope this is enough of a fix for now...

Reply:
Looks good to me now 👍🏻